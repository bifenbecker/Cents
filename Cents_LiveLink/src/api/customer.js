import httpClient from "./httpClient";

export const createCustomer = ({fullName, phoneNumber}) => {
  return httpClient({
    method: "POST",
    url: "/live-status/customer",
    data: {fullName, phoneNumber},
  });
};

export const verifyCustomer = ({phoneNumber}) => {
  return httpClient({
    method: "GET",
    url: "/live-status/customer/verify",
    params: {phoneNumber},
  });
};

export const addCustomerAddress = data => {
  return httpClient({
    method: "POST",
    url: "/live-status/customer/address/create",
    data,
  });
};

export const updateCustomerAddress = data => {
  return httpClient({
    method: "PATCH",
    url: "/live-status/customer/address/update",
    data,
  });
};

export const updateCustomerPhoneNumber = data => {
  return httpClient({
    method: "PATCH",
    url: "/live-status/customer/phone/update",
    data,
  });
};

export const saveDeliveryInstructions = data => {
  return httpClient({
    method: "PATCH",
    url: "/live-status/customer/address/instructions/update",
    data,
  });
};

export const getCustomerPaymentMethods = centsCustomerId => {
  return httpClient({
    method: "GET",
    url: `/live-status/customer/payment/${centsCustomerId}/payment-methods`,
  });
};

export const addCustomerPaymentMethod = data => {
  return httpClient({
    method: "POST",
    url: "/live-status/payment-methods/create",
    data,
  });
};

export const updateCustomerNotes = (storeCustomerId, data) => {
  return httpClient({
    method: "PATCH",
    url: `/live-status/customer/notes/${storeCustomerId}/update`,
    data,
  });
};
