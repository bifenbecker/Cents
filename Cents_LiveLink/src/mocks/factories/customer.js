export const VerifiedCustomer = {
  id: 1,
  firstName: "first",
  lastName: "last",
  phoneNumber: "1234567890",
  otp: "111111",
};

export const UnVerifiedCustomer = {
  id: 2,
  firstName: "first",
  lastName: "last",
  phoneNumber: "9999999999",
  otp: "222222",
};

export const ErrorCustomer = {
  id: 3,
  firstName: "first",
  lastName: "last",
  phoneNumber: "1111111111",
};
