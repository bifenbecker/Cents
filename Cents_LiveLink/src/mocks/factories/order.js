export const orderToken =
  "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MTQwODMsImlhdCI6MTY1MDUyNjE2N30.Rfbjm2XdRxK-x1NaUxdOYlSG0XOkN468DL4JvstQm9c";

export const orderId = 14083;

export const timeZone = "America/New_York";
