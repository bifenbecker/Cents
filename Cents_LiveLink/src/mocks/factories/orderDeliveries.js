export const ownDriverDelivery = {
  address1: "40 Lodge Street",
  address2: "apt",
  cancellationReason: null,
  centsCustomerAddressId: 325,
  city: "Albany",
  countryCode: "US",
  courierTip: 0,
  customerEmail: null,
  customerName: "Aditi 3 l",
  customerPhoneNumber: "4706659483",
  deliveredAt: null,
  deliveryProvider: "OWN_DRIVER",
  deliveryWindow: ["1650546000000", "1650564000000"],
  firstLevelSubdivisionCode: "NY",
  id: 2654,
  instructions: {leaveAtDoor: true, instructions: "call upon delivery"},
  orderId: 15025,
  postalCode: "12207",
  serviceFee: null,
  status: "SCHEDULED",
  storeCustomerId: 4490,
  storeId: 248,
  subsidyInCents: 0,
  thirdPartyDeliveryCostInCents: 0,
  thirdPartyDeliveryId: null,
  timingsId: 2125,
  totalDeliveryCost: 1,
  trackingUrl: null,
  type: "RETURN",
};

export const ownDriverPickup = {
  address1: "40 Lodge Street",
  address2: "apt",
  cancellationReason: null,
  centsCustomerAddressId: 325,
  city: "Albany",
  countryCode: "US",
  courierTip: 0,
  customerEmail: null,
  customerName: "Aditi 3 l",
  customerPhoneNumber: "4706659483",
  deliveredAt: null,
  deliveryProvider: "OWN_DRIVER",
  deliveryWindow: ["1650528000000", "1650546000000"],
  firstLevelSubdivisionCode: "NY",
  id: 2649,
  instructions: {leaveAtDoor: true, instructions: "call upon delivery"},
  orderId: 15025,
  postalCode: "12207",
  serviceFee: null,
  status: "SCHEDULED",
  storeCustomerId: 4490,
  storeId: 248,
  subsidyInCents: 0,
  thirdPartyDeliveryCostInCents: 0,
  thirdPartyDeliveryId: null,
  timingsId: 2118,
  totalDeliveryCost: 1,
  trackingUrl: null,
  type: "PICKUP",
};

export const onDemandDelivery = {
  address1: "40 Lodge Street",
  address2: "apt",
  cancellationReason: null,
  centsCustomerAddressId: 325,
  city: "Albany",
  countryCode: "US",
  courierTip: 3,
  customerEmail: null,
  customerName: "Aditi 3 l",
  customerPhoneNumber: "4706659483",
  deliveredAt: null,
  deliveryProvider: "DOORDASH",
  deliveryWindow: ["1650628800000", "1650630600000"],
  firstLevelSubdivisionCode: "NY",
  id: 2686,
  instructions: {leaveAtDoor: true, instructions: "call upon delivery"},
  orderId: 15065,
  postalCode: "12207",
  serviceFee: null,
  status: "INTENT_CREATED",
  storeCustomerId: 4490,
  storeId: 248,
  subsidyInCents: 500,
  thirdPartyDeliveryCostInCents: 999,
  thirdPartyDeliveryId: null,
  timingsId: 2140,
  totalDeliveryCost: 4.99,
  trackingUrl: null,
  type: "RETURN",
};

export const onDemandPickup = {
  address1: "40 Lodge Street",
  address2: "apt",
  cancellationReason: null,
  centsCustomerAddressId: 325,
  city: "Albany",
  countryCode: "US",
  courierTip: 3,
  customerEmail: null,
  customerName: "Aditi 3 l",
  customerPhoneNumber: "4706659483",
  deliveredAt: null,
  deliveryProvider: "DOORDASH",
  deliveryWindow: ["1650621600000", "1650623400000"],
  firstLevelSubdivisionCode: "NY",
  id: 2685,
  instructions: {leaveAtDoor: true, instructions: "call upon delivery"},
  orderId: 15065,
  postalCode: "12207",
  serviceFee: null,
  status: "INTENT_CREATED",
  storeCustomerId: 4490,
  storeId: 248,
  subsidyInCents: 500,
  thirdPartyDeliveryCostInCents: 999,
  thirdPartyDeliveryId: null,
  timingsId: 2140,
  totalDeliveryCost: 4.99,
  trackingUrl: null,
  type: "PICKUP",
};
