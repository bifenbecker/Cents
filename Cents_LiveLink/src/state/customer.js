import {createState} from "@hookstate/core";

export const initialCustomerState = {
  customerAuthToken: null,
  customer: {},
};

const customerState = createState(initialCustomerState);

export default customerState;
