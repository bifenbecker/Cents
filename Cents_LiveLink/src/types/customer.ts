export interface IStoreCustomers {
  notes: null;
  isHangDrySelected: boolean;
  hangDryInstructions: null;
  availableCredits: number | null;
  id: number;
  businessCustomer: {isCommercial: null};
}

export interface IPaymentMethods {
  last4: string;
  brand: string;
  centsCustomerId: number;
  provider: string;
  type: string;
  paymentMethodToken: string;
  id: number;
}

export interface IAddresses {
  id: number;
  address1: string;
  address2: string;
  city: string;
  firstLevelSubdivisionCode: string;
  postalCode: string;
  countryCode: string;
  googlePlacesId: string;
  instructions: string;
  leaveAtDoor: boolean;
}

export interface ICustomer {
  firstName: string;
  lastName: string;
  phoneNumber: string;
  id: number;
  storeCustomers: IStoreCustomers[];
  paymentMethods: IPaymentMethods[];
  addresses: IAddresses[];
  availableCredits: number;
}
