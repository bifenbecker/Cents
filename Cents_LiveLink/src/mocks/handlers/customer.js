import {rest} from "msw";
import {VerifiedCustomer} from "../factories/customer";

const customerHandlers = [
  rest.get(
    `${process.env.REACT_APP_BASE_URL}live-status/customer/verify`,
    (req, res, ctx) => {
      const phoneNumber = req.url.searchParams.get("phoneNumber");

      return res(
        phoneNumber === VerifiedCustomer.phoneNumber
          ? ctx.json({
              isVerified: true,
              firstName: VerifiedCustomer.firstName,
            })
          : ctx.json({isVerified: false})
      );
    }
  ),
];

export default customerHandlers;
