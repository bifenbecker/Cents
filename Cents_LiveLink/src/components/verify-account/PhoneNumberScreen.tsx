import React, {
  ChangeEvent,
  Dispatch,
  SetStateAction,
  useCallback,
  useEffect,
  useState,
} from "react";
import {Box, Button, Flex, Image, Text} from "rebass/styled-components";
import {toast} from "react-toastify";

import phoneNumberScreenTestids from "../../mocks/testids/PhoneNumberScreen.testids";

import {ExitIcon, IllustrationPhone} from "../../assets/images";
import {verifyCustomer} from "../../api/customer";
import {requestOtp} from "../../api/auth";
import {MaskedInput, TextField} from "../common";
import ToastError from "../common/ToastError";
import {onlineOrderState} from "../../state/online-order";
import {useHookstate} from "@hookstate/core";
import {NewCustomer} from "./types";
import {NOT_DIGITS_REGEXP, PHONE_LENGTH} from "../../constants/constants";

interface PhoneNumberScreenProps {
  phoneNumber: string;
  setPhoneNumber: Dispatch<SetStateAction<string>>;
  setLoading: Dispatch<SetStateAction<boolean>>;
  setOTP: Dispatch<SetStateAction<string>>;
  storeId: number | null;
  onlineOrderBusinessId: number | null;
  onSendVerificationClick: (newCustomerData: NewCustomer | null) => void;
}

interface VerifiedCustomer {
  firstName: string | null;
  lastName: string | null;
  isVerified: boolean;
}

const PhoneNumberScreen = ({
  onlineOrderBusinessId,
  setPhoneNumber,
  phoneNumber,
  onSendVerificationClick,
  setLoading,
  setOTP,
  storeId,
}: PhoneNumberScreenProps) => {
  const isAuthorized = useHookstate(onlineOrderState.isAuthorized).value;

  const [verifiedCustomer, setVerifiedCustomer] = useState<VerifiedCustomer | null>(null);

  const [{firstName, lastName}, setState] = useState<
    Omit<VerifiedCustomer, "isVerified">
  >({
    firstName: "",
    lastName: "",
  });

  const verifyPhoneNumber = useCallback(
    async (value: string) => {
      try {
        setLoading(true);
        const {data} = await verifyCustomer({phoneNumber: value});
        setVerifiedCustomer(data ?? {});
      } catch (error) {
        toast.error(
          <ToastError
            message={"Could not check your account linked to your phone number"}
          />
        );
      } finally {
        setLoading(false);
      }
    },
    [setLoading]
  );

  useEffect(() => {
    if (phoneNumber?.length === PHONE_LENGTH) {
      verifyPhoneNumber(phoneNumber);
    }
  }, [verifyPhoneNumber, phoneNumber]);

  const onPhoneNumberChange = ({target: {value}}: ChangeEvent<HTMLInputElement>) => {
    setVerifiedCustomer(null);
    setState(prevState => ({...prevState, lastName: "", firstName: ""}));
    setPhoneNumber(value.replace(NOT_DIGITS_REGEXP, "").slice(0, PHONE_LENGTH) as string);
  };

  const sendVerificationHandler = async () => {
    onSendVerificationClick(
      verifiedCustomer?.isVerified
        ? null
        : {fullName: `${firstName} ${lastName}`.trim(), phoneNumber}
    );
    if (isAuthorized) {
      setLoading(true);
      try {
        const res = await requestOtp({phoneNumber, storeId, isAuthorized});
        setOTP(res.data.otpCode);
      } catch (error) {
        console.log("err: ", error);
      } finally {
        setLoading(false);
      }
    }
  };

  const onHandleChange = ({
    target: {value, name},
  }: ChangeEvent<HTMLTextAreaElement & HTMLInputElement>) => {
    const trimmedValue = value.trim().replace(/\s+/g, " ");
    setState(prevState => ({...prevState, [name]: trimmedValue}));
  };

  return (
    <>
      <Text fontSize={24} mb={20} textAlign="center">
        What’s your phone number?
      </Text>
      <MaskedInput
        onChange={onPhoneNumberChange}
        value={phoneNumber || ""}
        mask="( 9 9 9 ) 9 9 9 - 9 9 9 9"
        type="tel"
        {...styles.phoneInput}
      />
      {!verifiedCustomer && onlineOrderBusinessId && (
        <Text
          fontFamily="secondary"
          pt="20px"
          px="16px"
          fontSize="16px"
          style={styles.text}
        >
          We’ll only use it to verify your identity and provide updates about your order.
        </Text>
      )}
      {verifiedCustomer && (
        <Flex {...styles.verifiedCustomer.wrapper}>
          {!verifiedCustomer?.isVerified && (
            <>
              <TextField
                label="First Name"
                name="firstName"
                onChange={onHandleChange}
                value={firstName || ""}
                suffix={
                  <Image
                    src={ExitIcon}
                    height="16px"
                    sx={{cursor: "pointer"}}
                    onClick={() => setState(prevState => ({...prevState, lastName: ""}))}
                  />
                }
                materialWrapperStyle={{width: "100%", height: "56px", marginBottom: "4%"}}
                wrapperInputStyle={{fontSize: "18px"}}
                data-testid={phoneNumberScreenTestids.firstNameInput}
              />
              <TextField
                label="Last Name"
                name="lastName"
                onChange={onHandleChange}
                value={lastName || ""}
                suffix={
                  <Image
                    src={ExitIcon}
                    height="16px"
                    sx={{cursor: "pointer"}}
                    onClick={() => setState(prevState => ({...prevState, lastName: ""}))}
                  />
                }
                materialWrapperStyle={{width: "100%", height: "56px"}}
                wrapperInputStyle={{fontSize: "18px"}}
                data-testid={phoneNumberScreenTestids.lastNameInput}
              />
            </>
          )}
          <Flex {...styles.verifiedCustomer.verification.wrapper}>
            <Flex {...styles.verifiedCustomer.verification.text}>
              <Box mr="32px">
                <Text fontSize="18px">
                  {verifiedCustomer?.isVerified
                    ? `Welcome back, ${verifiedCustomer?.firstName}`
                    : "Verify Your Account"}
                </Text>
                <br />
                <Text fontFamily="secondary" fontSize="14px">
                  {verifiedCustomer?.isVerified
                    ? "Verify your account to login"
                    : "We’ll send you a text with a code to create your account."}
                </Text>
              </Box>
              <Image src={IllustrationPhone} flexShrink="0" />
            </Flex>
            <Button
              variant="primary"
              {...styles.verifiedCustomer.verification.button}
              onClick={sendVerificationHandler}
              disabled={
                (!verifiedCustomer?.isVerified && (!firstName || !lastName)) ||
                (isAuthorized ? false : Boolean(isAuthorized))
              }
              data-testid={phoneNumberScreenTestids.sendVerificationButton}
            >
              {isAuthorized ? "SHOW VERIFICATION CODE" : "SEND VERIFICATION CODE"}
            </Button>
          </Flex>
        </Flex>
      )}
    </>
  );
};

const styles = {
  phoneInput: {
    sx: {textAlign: ["center", "center", "center"]},
    fontSize: ["1.5rem", "1.75rem", "2rem"],
    px: 0,
    width: ["100%", "100%", "100%", "75%"],
  },
  verifiedCustomer: {
    wrapper: {
      width: ["100%", "100%", "100%", "50%"],
      maxWidth: "320px",
      flexDirection: "column" as any,
      mt: "1.5rem",
      alignItems: "center",
      justifyContent: "center",
    },
    verification: {
      wrapper: {
        width: "100%",
        p: "1.25rem",
        my: "1.5rem",
        flexDirection: "column" as any,
        bg: "WHITE",
        sx: {
          borderRadius: "24px",
          boxShadow:
            "0 4px 5px 0 rgba(0,0,0,0.14), 0 1px 10px 0 rgba(0,0,0,0.12), 0 2px 4px -1px rgba(0,0,0,0.2)",
        },
      },
      text: {
        justifyContent: "space-between",
        height: "83px",
        alignItems: "center",
      },
      button: {
        mt: "1rem",
        height: "48px",
        px: "24px",
      },
    },
    disclaimerText: {
      sx: {
        fontFamily: "secondary",
        fontStyle: "italic",
      },
      pt: "20px",
    },
  },
  text: {
    textAlign: "center" as any,
  },
};

export default PhoneNumberScreen;
