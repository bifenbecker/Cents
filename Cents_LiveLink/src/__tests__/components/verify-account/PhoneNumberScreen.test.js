import React from "react";
import {render, waitFor, fireEvent} from "@testing-library/react";

import {VerifiedCustomer, UnVerifiedCustomer} from "../../../mocks/factories/customer";
import phoneNumberScreenTestids from "../../../mocks/testids/PhoneNumberScreen.testids";

import PhoneNumberScreen from "../../../components/verify-account/PhoneNumberScreen";

const renderPhoneNumberScreen = props => {
  const {getByTestId, findByTestId, queryByTestId, getByText} = render(
    <PhoneNumberScreen {...props} />
  );

  return {getByTestId, findByTestId, queryByTestId, getByText};
};

describe("PhoneNumberScreen Component", () => {
  // TODO: Check if we can manipulate MaskedInput used for phoneNumber.

  describe("When no phoneNumber is given", () => {
    it("should match snapshot", () => {
      expect(
        render(<PhoneNumberScreen phoneNumber={""} setLoading={jest.fn()} />)
      ).toMatchSnapshot();
    });

    it("should not display button or first name or last name input", async () => {
      const {queryByTestId} = renderPhoneNumberScreen({
        phoneNumber: "",
        setLoading: jest.fn(),
        onSendVerificationClick: jest.fn(),
      });
      expect(queryByTestId(phoneNumberScreenTestids.sendVerificationButton)).toBeNull();
      expect(queryByTestId(phoneNumberScreenTestids.firstNameInput)).toBeFalsy();
      expect(queryByTestId(phoneNumberScreenTestids.lastNameInput)).toBeFalsy();
    });
  });

  describe("When verified 10 digit phoneNumber is given", () => {
    it("should match snapshot", () => {
      expect(
        render(
          <PhoneNumberScreen
            phoneNumber={VerifiedCustomer.phoneNumber}
            setLoading={jest.fn()}
          />
        )
      ).toMatchSnapshot();
    });

    it("should display welcome back text and verification button", async () => {
      const {getByText} = renderPhoneNumberScreen({
        phoneNumber: VerifiedCustomer.phoneNumber,
        setLoading: jest.fn(),
      });

      await waitFor(() =>
        expect(getByText(`Welcome back, ${VerifiedCustomer.firstName}`)).not.toBeNull()
      );

      await waitFor(() =>
        expect(getByText("Verify your account to login")).not.toBeNull()
      );

      await waitFor(() => expect(getByText("SEND VERIFICATION CODE")).not.toBeNull());
    });

    it("should call onSendVerificationClick with no args when submit btn is clicked", async () => {
      const onSendVerificationClick = jest.fn();
      const {findByTestId} = renderPhoneNumberScreen({
        phoneNumber: VerifiedCustomer.phoneNumber,
        setLoading: jest.fn(),
        onSendVerificationClick: onSendVerificationClick,
      });

      const button = await findByTestId(phoneNumberScreenTestids.sendVerificationButton);

      fireEvent.click(button);

      expect(onSendVerificationClick).toBeCalledTimes(1);
      expect(onSendVerificationClick).toBeCalledWith(null);
    });
    // TODO: If we can manipulate MaskedInput used for phoneNumber,
    // changing the value less than 10 digits shold take the customer back to initial stage.
    describe("When the admin/manager is authorised", () => {});
  });

  describe("When un-verified 10 digit phoneNumber is given", () => {
    it("should match snapshot", () => {
      expect(
        render(
          <PhoneNumberScreen
            phoneNumber={UnVerifiedCustomer.phoneNumber}
            setLoading={jest.fn()}
          />
        )
      ).toMatchSnapshot();
    });

    it("should display verify your phone number and verification button", async () => {
      const {getByText} = renderPhoneNumberScreen({
        phoneNumber: UnVerifiedCustomer.phoneNumber,
        setLoading: jest.fn(),
      });

      await waitFor(() => expect(getByText("Verify Your Account")).not.toBeNull());

      await waitFor(() =>
        expect(
          getByText("Weâ€™ll send you a text with a code to create your account.")
        ).not.toBeNull()
      );

      await waitFor(() => expect(getByText("SEND VERIFICATION CODE")).not.toBeNull());
    });

    it("SEND VERIFICATION CODE should be disabled if first name and last name were empty", async () => {
      const {findByTestId} = renderPhoneNumberScreen({
        phoneNumber: UnVerifiedCustomer.phoneNumber,
        setLoading: jest.fn(),
      });
      const button = await findByTestId(phoneNumberScreenTestids.sendVerificationButton);
      expect(button).toBeDisabled();
    });

    it("SEND VERIFICATION CODE should be disabled if either of first name or last name were empty", async () => {
      const onSendVerificationClick = jest.fn();
      const {findByTestId, getByTestId} = renderPhoneNumberScreen({
        phoneNumber: UnVerifiedCustomer.phoneNumber,
        setLoading: jest.fn(),
        onSendVerificationClick: onSendVerificationClick,
      });
      const button = await findByTestId(phoneNumberScreenTestids.sendVerificationButton);
      fireEvent.change(getByTestId(phoneNumberScreenTestids.firstNameInput), {
        target: {value: UnVerifiedCustomer.firstName},
      });
      expect(button).toBeDisabled();
    });

    it("should call onSendVerificationClick with args when submit btn is clicked", async () => {
      const onSendVerificationClick = jest.fn();
      const {findByTestId, getByTestId} = renderPhoneNumberScreen({
        phoneNumber: UnVerifiedCustomer.phoneNumber,
        setLoading: jest.fn(),
        onSendVerificationClick: onSendVerificationClick,
      });
      const button = await findByTestId(phoneNumberScreenTestids.sendVerificationButton);
      fireEvent.change(getByTestId(phoneNumberScreenTestids.firstNameInput), {
        target: {value: UnVerifiedCustomer.firstName},
      });
      fireEvent.change(getByTestId(phoneNumberScreenTestids.lastNameInput), {
        target: {value: UnVerifiedCustomer.lastName},
      });
      fireEvent.click(button);
      expect(onSendVerificationClick).toBeCalledTimes(1);
      expect(onSendVerificationClick).toBeCalledWith({
        fullName: `${UnVerifiedCustomer.firstName} ${UnVerifiedCustomer.lastName}`.trim(),
        phoneNumber: UnVerifiedCustomer.phoneNumber,
      });
    });
    // TODO: If we can manipulate MaskedInput used for phoneNumber,
    // changing the value less than 10 digits shold take the customer back to initial stage.
    describe("When the admin/manager is authorised", () => {});
  });
});
