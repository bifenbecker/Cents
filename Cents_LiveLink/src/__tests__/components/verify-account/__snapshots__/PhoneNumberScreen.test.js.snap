// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneNumberScreen Component When no phoneNumber is given should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ dyPies"
        font-size="24"
      >
        What’s your phone number?
      </div>
      <input
        class="sc-bczRLJ daJrKh"
        font-size="1.5rem,1.75rem,2rem"
        font-weight="600"
        type="tel"
        value="( _ _ _ ) _ _ _ - _ _ _ _"
        width="100%,100%,100%,75%"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ dyPies"
      font-size="24"
    >
      What’s your phone number?
    </div>
    <input
      class="sc-bczRLJ daJrKh"
      font-size="1.5rem,1.75rem,2rem"
      font-weight="600"
      type="tel"
      value="( _ _ _ ) _ _ _ - _ _ _ _"
      width="100%,100%,100%,75%"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`PhoneNumberScreen Component When un-verified 10 digit phoneNumber is given should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ dyPies"
        font-size="24"
      >
        What’s your phone number?
      </div>
      <input
        class="sc-bczRLJ daJrKh"
        font-size="1.5rem,1.75rem,2rem"
        font-weight="600"
        type="tel"
        value="( 9 9 9 ) 9 9 9 - 9 9 9 9"
        width="100%,100%,100%,75%"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ dyPies"
      font-size="24"
    >
      What’s your phone number?
    </div>
    <input
      class="sc-bczRLJ daJrKh"
      font-size="1.5rem,1.75rem,2rem"
      font-weight="600"
      type="tel"
      value="( 9 9 9 ) 9 9 9 - 9 9 9 9"
      width="100%,100%,100%,75%"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`PhoneNumberScreen Component When verified 10 digit phoneNumber is given should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ dyPies"
        font-size="24"
      >
        What’s your phone number?
      </div>
      <input
        class="sc-bczRLJ daJrKh"
        font-size="1.5rem,1.75rem,2rem"
        font-weight="600"
        type="tel"
        value="( 1 2 3 ) 4 5 6 - 7 8 9 0"
        width="100%,100%,100%,75%"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ dyPies"
      font-size="24"
    >
      What’s your phone number?
    </div>
    <input
      class="sc-bczRLJ daJrKh"
      font-size="1.5rem,1.75rem,2rem"
      font-weight="600"
      type="tel"
      value="( 1 2 3 ) 4 5 6 - 7 8 9 0"
      width="100%,100%,100%,75%"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
